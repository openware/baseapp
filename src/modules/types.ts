import { MarketsState, TickerEvent } from './public/markets';
import { OrderBookState } from './public/orderBook';
import { OrdersState } from './user/orders';

export interface CommonState {
    error?: CommonError;
    loading?: boolean;
}

export type OrderStatus = 'wait' | 'done' | 'cancel' | 'pending' | 'reject';
export type OrderSide = 'sell' | 'buy';
export type OrderType = 'limit' | 'market';
export type OrderKind = 'bid' | 'ask';

export interface OrderCommon {
    price: number;
    created_at: string;
    state: OrderStatus;
    remaining_volume: number;
    origin_volume: number;
    executed_volume: number;
    side: OrderSide;
    market: string;
    ord_type?: OrderType;
    avg_price?: number;
    volume?: number;
    updated_at?: string;
    confirmed?: boolean;
    uuid?: string;
    id?: number;
}

/*
** example: {"id":10666,"side":"buy","ord_type":"limit","price":"0.003","avg_price":"0.003","state":"wait","market":"kyneth","created_at":"2019-02-15T09:46:21+01:00","volume":"10.0","remaining_volume":"9.9","executed_volume":"0.1","trades_count":1}
*/
export interface OrderAPI {
    side: OrderSide;
    ord_type: OrderType;
    price: string;
    state: OrderStatus;
    market: string;
    created_at: string;
    remaining_volume: string;
    executed_volume: string;
    origin_volume: string;
    avg_price: string;
    updated_at?: string;
    confirmed?: boolean;
    uuid?: string;
    id?: number;
}

/*
** example: {"order":{"id":10666,"at":1550220381,"market":"kyneth","kind":"bid","price":"0.003","state":"wait","volume":"9.9","origin_volume":"10.0"}}
*/
export interface OrderEvent {
    at: number;
    market: string;
    kind: OrderKind;
    price: string;
    state: OrderStatus;
    origin_volume: string;
    remaining_volume: string;
    ord_type?: OrderType;
    updated_at?: string;
    confirmed?: boolean;
    uuid?: string;
    id?: number;
}


export interface MarketUpdateEvent {
    asks: Array<[number, number]>;
    bids: Array<[number, number]>;
}

export type RangerEvent = TickerEvent | OrderEvent | MarketUpdateEvent;

export interface CommonError {
    code: number;
    message: string[];
}

export interface CoreState {
    orders: OrdersState;
    orderBook: OrderBookState;
    markets: MarketsState;
}

export interface ClientState {
    coreData: CoreState;
}
