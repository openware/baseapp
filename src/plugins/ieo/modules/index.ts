export * from './public';
export * from './user';
export * from './app';
