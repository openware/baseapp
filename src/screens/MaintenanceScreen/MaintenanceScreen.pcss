.pg-maintenance-screen {
  display: flex;
  height: 100vh;
  justify-content: space-around;
  width: 100%;

  &__container {
    text-align: center;

    &-header {
      padding: 80px 0 120px 0;
    }

    &-body {
      &-title {
        color: var(--primary-text-color);
        font-family: var(--font-family);
        font-size: 48px;
        font-weight: 500;
        line-height: 56px;
        max-width: 500px;
        padding: 16px;
      }

      &-subtitle {
        color: var(--primary-text-color);
        font-family: var(--font-family);
        font-size: 22px;
        font-weight: 300;
        padding-top: 10px;
      }
    }
  }

  &__background {
    position: absolute;
    top: 50px;
  }
}
