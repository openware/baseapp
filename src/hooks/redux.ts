import { useSelector } from 'react-redux';
import { RootState } from '../modules';

export function useReduxState<T>(callback: (state: RootState) => T): T {
    return useSelector(callback);
}
