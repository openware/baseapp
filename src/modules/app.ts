import { combineReducers } from 'redux';
import { alertReducer  } from './public/alert';
import { changeColorThemeReducer  } from './public/colorTheme';
import { configsReducer } from './public/configs';
import { currenciesReducer } from './public/currencies';
import { gridLayoutReducer } from './public/gridLayout/reducer';
import { changeLanguageReducer  } from './public/i18n';
import { klineReducer  } from './public/kline';
import { marketsReducer } from './public/markets';
import { memberLevelsReducer } from './public/memberLevels';
import {
    depthReducer,
    incrementDepthReducer,
    orderBookReducer,
} from './public/orderBook';
import { rangerReducer  } from './public/ranger/reducer';
import { recentTradesReducer  } from './public/recentTrades';
import { apiKeysReducer } from './user/apiKeys';
import { authReducer  } from './user/auth';
import { beneficiariesReducer } from './user/beneficiaries';
import { getGeetestCaptchaReducer } from './user/captcha';
import { sendEmailVerificationReducer } from './user/emailVerification';
import { historyReducer  } from './user/history';
import {
    documentsReducer,
    identityReducer,
    labelReducer,
    phoneReducer,
} from './user/kyc';
import { newHistoryReducer } from './user/newHistory';
import { openOrdersReducer } from './user/openOrders';
import { ordersReducer  } from './user/orders';
import { ordersHistoryReducer  } from './user/ordersHistory';
import { passwordReducer  } from './user/password';
import { profileReducer  } from './user/profile';
import { userActivityReducer  } from './user/userActivity';
import { walletsReducer  } from './user/wallets';
import { withdrawLimitReducer  } from './user/withdrawLimit';

export const publicReducer = combineReducers({
    colorTheme: changeColorThemeReducer,
    configs: configsReducer,
    currencies: currenciesReducer,
    recentTrades: recentTradesReducer,
    markets: marketsReducer,
    orderBook: orderBookReducer,
    depth: depthReducer,
    incrementDepth: incrementDepthReducer,
    ranger: rangerReducer,
    i18n: changeLanguageReducer,
    kline: klineReducer,
    alerts: alertReducer,
    rgl: gridLayoutReducer,
    memberLevels: memberLevelsReducer,
});

export const userReducer = combineReducers({
    auth: authReducer,
    beneficiaries: beneficiariesReducer,
    label: labelReducer,
    orders: ordersReducer,
    password: passwordReducer,
    profile: profileReducer,
    wallets: walletsReducer,
    phone: phoneReducer,
    identity: identityReducer,
    documents: documentsReducer,
    history: historyReducer,
    newHistory: newHistoryReducer,
    apiKeys: apiKeysReducer,
    userActivity: userActivityReducer,
    ordersHistory: ordersHistoryReducer,
    openOrders: openOrdersReducer,
    sendEmailVerification: sendEmailVerificationReducer,
    captchaKeys: getGeetestCaptchaReducer,
    withdrawLimit: withdrawLimitReducer,
});
