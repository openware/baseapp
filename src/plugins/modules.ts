import { combineReducers } from 'redux';
// tslint:disable-next-line no-submodule-imports
import { all, call } from 'redux-saga/effects';
import {
    ieoReducer,
    rootIEOSaga,
    StateIEO,
} from './ieo';

export interface PluginsState {
    ieo: StateIEO;
}

export const pluginsReducer = combineReducers({
    ieo: ieoReducer,
});

export function* rootPluginsSaga() {
    yield all([
        call(rootIEOSaga),
    ]);
}
