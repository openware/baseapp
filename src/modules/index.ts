import { combineReducers } from 'redux';
// tslint:disable-next-line no-submodule-imports
import { all, call } from 'redux-saga/effects';
import { publicReducer, userReducer } from './app';
import { AlertState, rootHandleAlertSaga } from './public/alert';
import { ColorThemeState } from './public/colorTheme';
import { ConfigsState, rootConfigsSaga } from './public/configs';
import { CurrenciesState, rootCurrenciesSaga } from './public/currencies';
import { GridLayoutState } from './public/gridLayout';
import { LanguageState } from './public/i18n';
import { KlineState, rootKlineFetchSaga } from './public/kline';
import { MarketsState, rootMarketsSaga } from './public/markets';
import {
    DepthIncrementState,
    DepthState,
    OrderBookState,
    rootOrderBookSaga,
} from './public/orderBook';
import { RangerState } from './public/ranger/reducer';
import { RecentTradesState, rootRecentTradesSaga } from './public/recentTrades';
import { ApiKeysState } from './user/apiKeys';
import { rootApiKeysSaga } from './user/apiKeys/sagas';
import { AuthState, rootAuthSaga } from './user/auth';
import { BeneficiariesState, rootBeneficiariesSaga } from './user/beneficiaries';
import { GeetestCaptchaState, rootGeetestCaptchaSaga } from './user/captcha';
import { EmailVerificationState, rootEmailVerificationSaga } from './user/emailVerification';
import { HistoryState, rootHistorySaga } from './user/history';
import { DocumentsState, rootSendDocumentsSaga } from './user/kyc/documents';
import { IdentityState, rootSendIdentitySaga } from './user/kyc/identity';
import { LabelState, rootLabelSaga } from './user/kyc/label';
import { PhoneState, rootSendCodeSaga } from './user/kyc/phone';
import { NewHistoryState, rootNewHistorySaga } from './user/newHistory';
import { OpenOrdersState, rootOpenOrdersSaga } from './user/openOrders';
import { OrdersState, rootOrdersSaga } from './user/orders';
import { OrdersHistoryState, rootOrdersHistorySaga } from './user/ordersHistory';
import { PasswordState, rootPasswordSaga } from './user/password';
import { ProfileState, rootProfileSaga } from './user/profile';
import { rootUserActivitySaga, UserActivityState } from './user/userActivity';
import { rootWalletsSaga, WalletsState } from './user/wallets';
import { rootWithdrawLimitSaga, WithdrawLimitState } from './user/withdrawLimit';
import { MemberLevelsState, rootMemberLevelsSaga } from './public/memberLevels';

export * from './public/markets';
export * from './public/orderBook';
export * from './public/colorTheme';
export * from './public/configs';
export * from './public/currencies';
export * from './public/i18n';
export * from './public/kline';
export * from './public/alert';
export * from './user/apiKeys';
export * from './user/auth';
export * from './user/beneficiaries';
export * from './user/captcha';
export * from './user/wallets';
export * from './user/profile';
export * from './user/openOrders';
export * from './user/orders';
export * from './user/ordersHistory';
export * from './user/password';
export * from './user/userActivity';
export * from './user/history';
export * from './user/newHistory';
export * from './user/kyc';
export * from './user/emailVerification';
export * from './user/withdrawLimit';
export * from './public/memberLevels';

export interface RootState {
    public: {
        colorTheme: ColorThemeState;
        configs: ConfigsState;
        currencies: CurrenciesState;
        recentTrades: RecentTradesState;
        markets: MarketsState;
        orderBook: OrderBookState;
        depth: DepthState;
        incrementDepth: DepthIncrementState;
        ranger: RangerState;
        i18n: LanguageState;
        alerts: AlertState;
        kline: KlineState;
        rgl: GridLayoutState;
        memberLevels: MemberLevelsState;
    };
    user: {
        auth: AuthState;
        beneficiaries: BeneficiariesState;
        orders: OrdersState;
        password: PasswordState;
        profile: ProfileState;
        label: LabelState;
        wallets: WalletsState;
        documents: DocumentsState;
        identity: IdentityState;
        phone: PhoneState;
        history: HistoryState;
        newHistory: NewHistoryState;
        apiKeys: ApiKeysState;
        userActivity: UserActivityState;
        ordersHistory: OrdersHistoryState;
        openOrders: OpenOrdersState;
        sendEmailVerification: EmailVerificationState;
        captchaKeys: GeetestCaptchaState;
        withdrawLimit: WithdrawLimitState;
    };
}

export const rootReducer = combineReducers({
    public: publicReducer,
    user: userReducer,
});

export function* rootSaga() {
    yield all([
        call(rootApiKeysSaga),
        call(rootAuthSaga),
        call(rootBeneficiariesSaga),
        call(rootConfigsSaga),
        call(rootCurrenciesSaga),
        call(rootEmailVerificationSaga),
        call(rootGeetestCaptchaSaga),
        call(rootHandleAlertSaga),
        call(rootHistorySaga),
        call(rootKlineFetchSaga),
        call(rootLabelSaga),
        call(rootMarketsSaga),
        call(rootMemberLevelsSaga),
        call(rootNewHistorySaga),
        call(rootOpenOrdersSaga),
        call(rootOrderBookSaga),
        call(rootOrdersHistorySaga),
        call(rootOrdersSaga),
        call(rootPasswordSaga),
        call(rootProfileSaga),
        call(rootRecentTradesSaga),
        call(rootSendCodeSaga),
        call(rootSendDocumentsSaga),
        call(rootSendIdentitySaga),
        call(rootUserActivitySaga),
        call(rootWalletsSaga),
        call(rootWithdrawLimitSaga),
    ]);
}
