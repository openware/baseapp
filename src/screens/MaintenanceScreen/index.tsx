import * as React from 'react';
import { injectIntl } from 'react-intl';
import { connect, MapStateToProps } from 'react-redux';
import { RouterProps } from 'react-router';
import { withRouter } from 'react-router-dom';
import { compose } from 'redux';
import { IntlProps } from '../../';
import { BackgroundMaintenance } from '../../assets/images/BackgroundMaintenance';
import { LogoIcon } from '../../assets/images/LogoIcon';
import { setDocumentTitle } from '../../helpers';
import { RootState, selectPlatformAccessStatus } from '../../modules';

interface ReduxProps {
    status: string;
}

type Props = RouterProps & IntlProps & ReduxProps;

class Maintenance extends React.Component<Props> {
    public componentDidMount() {
        setDocumentTitle('500');
        if (this.props.status.length && this.props.status !== 'maintenance') {
            this.props.history.replace('/');
        }
    }

    public componentWillReceiveProps(nextProps: Props) {
        if (!this.props.status.length && nextProps.status.length && nextProps.status !== 'maintenance') {
            this.props.history.replace('/');
        }
    }

    public render() {
        return (
            <div className="pg-maintenance-screen">
                <div className="pg-maintenance-screen__container">
                    <div className="pg-maintenance-screen__container-header">
                        <LogoIcon />
                    </div>
                    <div className="pg-maintenance-screen__container-body">
                        <div className="pg-maintenance-screen__container-body-title">
                            {this.translate('page.body.500.maintenance')}
                        </div>
                        <div className="pg-maintenance-screen__container-body-subtitle">
                            {this.translate('page.body.500.availableSoon')}
                        </div>
                    </div>
                </div>
                <div className="pg-maintenance-screen__background">
                    <BackgroundMaintenance />
                </div>
            </div>
        );
    }

    private translate = (key: string) => this.props.intl.formatMessage({id: key});
}

const mapStateToProps: MapStateToProps<ReduxProps, {}, RootState> = state => ({
    status: selectPlatformAccessStatus(state),
});

export const MaintenanceScreen = compose(
    injectIntl,
    withRouter,
    connect(mapStateToProps),
)(Maintenance) as React.ComponentClass;
