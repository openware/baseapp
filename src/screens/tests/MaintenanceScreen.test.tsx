import { shallow } from 'enzyme';
import * as React from 'react';
import { InjectedIntlProps } from 'react-intl';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { rootReducer } from '../../modules';
import { MaintenanceScreen } from '../MaintenanceScreen';

const store = createStore(rootReducer);

const setup = (props: Partial<InjectedIntlProps> = {}) =>
    shallow(
        <Provider store={store}>
            <MaintenanceScreen />
        </Provider>,
    );

describe('MaintenanceScreen', () => {
    const wrapper = setup();

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
