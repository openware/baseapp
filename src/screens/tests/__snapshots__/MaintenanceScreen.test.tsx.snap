// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaintenanceScreen should render 1`] = `ShallowWrapper {}`;
